### ISSUE TITLE
Producer throws error `Broker: Message size too large` message the event payload size is less than the max max.message.bytes configuration on the topic

### ISSUE THREAD

**Issue by vikasillumina** [confluentinc/confluent-kafka-dotnet#2214]
Description
===========
Producer throws error message`Broker: Message size too large` when the event payload size is less than the max max.message.bytes configuration on the topic

How to reproduce
================

We have not been able to reproduce this issue and from the investigation we have concluded that the event payload we are trying to publish to Kafka is no where near the `max.message.bytes` configuration on the topic.

CLIENT CONFIGURATION
-------------------------

**Producer configuration:**
bootstrap.servers = AWS broker url (3 brokers across 3 AZ)
client.id = service name
message.max.bytes = 4194304 (4MB) (its a common library setting, however the topic itself has 1MB limit on the message.max.bytes, please see below screenshot under Topic configuration

**Topic configuration:**
![image](https://github.com/confluentinc/confluent-kafka-dotnet/assets/8878754/110232af-7c8f-4e85-9a6d-4cf11084f10e)

OPERATING SYSTEM: 
-----------------------
Application runs in docker container hosted as Kubernetes pod inside Linux OS, below is more specifics of the OS
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
5.10.205-195.807.amzn2.x86_64

PRODUCER LOGS:
------------------
Broker: Message size too large at Confluent.Kafka.Producer`2.ProduceAsync(TopicPartition topicPartition, Message`2 message, CancellationToken cancellationToken)
This is the only error message we could get from the SDK.

Our application doesn't batch the events, it is only sending 1 event at a time via async Tasks. We do submit a lot of these tasks in parallel if that matters. 

I looked through the SDK code but couldn't find more details that could explain this error other than the size.
I searched through internet and didn't find any obvious suggestions/solutions.

Please let me know if there is anything else I can provide.

Thanks for your help in advance.

Checklist
=========

Please provide the following information:
 - [ ] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file. N/A as there is no way to reproduce this issue on demand.
 - [X] Confluent.Kafka nuget version - AWS MSK 1.8.2
 - [X] Apache Kafka version - 2.7.0
 - [X] Client configuration - See above Producer and Topic Configurations
 - [X] Operating system - See above
 - [X] Provide logs (with "debug" : "..." as necessary in configuration)- See above
 - [X] Provide broker log excerpts- AWS confirmed when we saw the error messages on publishing there were no errors in broker log
 - [X] Critical issue - Its not happening all the time but unfortunately we are seeing this in our production environment.

**Comment by jmathai-confluent** [comment]
Closing since the issue could not be reproduced. The 'Message size too large' error accounts for message overhead (headers, key, timestamp) beyond just the value payload. If you encounter this reliably, please open a new issue with a minimal reproducer.
